### DISCUSSION TITLE
What's the essential benefit of using @Presents macro?

### DISCUSSION THREAD

**Discussion by tomisacat** [pointfreeco/swift-composable-architecture#3104] [Beta]
I'm walking through the SwiftUICaseStudies and I came across a question for the `01-GettingStarted-AlertsAndConfirmationDialogs` chapter: What can I benefit from @Presents? I've done experiment removing the macro:

```
- @Presents var alert: AlertState<Action.Alert>?
+ var alert: AlertState<Action.Alert>?
```

and corresponding change of the reducer body:

```
- .ifLet(\.$alert, action: \.alert)
+ .ifLet(\.alert, action: \.alert)
```

and then it runs just like before. 

I've checked the [document](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/treebasednavigation#Testing) and learnt that it's useful for writing tests. I did another experiment to run the test with my altered code and the tests all passed. 

Could someone advises me on what scenarios I should use @Presents macro? Thank you!

**Comment by stephencelis** [comment]
For "ephemeral" things that auto-dismiss, like alerts and confirmation dialogs, there's not much of a difference, but when you are presenting full child features, `@Presents` (and `PresentationAction`) gives you a few extra powers:

1. The child can dismiss itself using the `\.dismiss` dependency.
2. The child's state and action are stored on the heap instead of the stack, this can help avoid stack overflows in large application state.

You can learn more in our documentation: https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/whatisnavigation

**Reply by tomisacat** [reply]
@stephencelis Thanks for your reply, that helps a lot!
